### PR TITLE
btcm: add modify order functionality, change return to pointer

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Adds modify order functionality to wrapper. 
Modifies return across the exchange interface to return a pointer of order.Modify. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/btcmarkets/btcmarkets.go [hunk 1]
```diff
@@ -483,6 +483,17 @@ func (b *BTCMarkets) RemoveOrder(ctx context.Context, id string) (CancelOrderRes
 		request.Auth)
 }
 
+// Replace removes a given order
+func (b *BTCMarkets) ReplaceAnOrder(ctx context.Context, id string, price, amount float64) (*OrderData, error) {
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets.go:487]: Test Parallel

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 2]
```diff
@@ -571,8 +572,42 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (order.Su
 
 // ModifyOrder will allow of changing orderbook placement and limit to
 // market conversion
-func (b *BTCMarkets) ModifyOrder(ctx context.Context, action *order.Modify) (order.Modify, error) {
-	return order.Modify{}, common.ErrFunctionNotSupported
+func (b *BTCMarkets) ModifyOrder(ctx context.Context, action *order.Modify) (*order.Modify, error) {
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:575]: Test Parallel

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 3]
```diff
@@ -571,8 +572,42 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (order.Su
...
+	err := action.Validate()
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:576]: Strange linting issue. Doesn't really seem necessary, happy with a nolint

File: exchanges/interfaces.go [hunk 4]
```diff
@@ -100,6 +97,16 @@ type IBotExchange interface {
...
 }
 
+// OrderManagement defines functionality for order management
+type OrderManagement interface {
+	SubmitOrder(ctx context.Context, s *order.Submit) (order.SubmitResponse, error)
+	ModifyOrder(ctx context.Context, action *order.Modify) (*order.Modify, error)
+	CancelOrder(ctx context.Context, o *order.Cancel) error
+	CancelBatchOrders(ctx context.Context, o []order.Cancel) (order.CancelBatchResponse, error)
+	CancelAllOrders(ctx context.Context, orders *order.Cancel) (order.CancelAllResponse, error)
+	GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, assetType asset.Item) (order.Detail, error)
```
Comment (gloriousCode) [inline exchanges/interfaces.go:107]: I feel like GetOrderHistorty & GetActiveOrders belong here

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice

File: exchanges/btcmarkets/btcmarkets.go (hunk 1 above)
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets.go:487]: Please add `clientOrderId` support

Comment (shazbert) [reply exchanges/btcmarkets/btcmarkets.go:487]: How bout I also add the price and amount as well. 😆

File: exchanges/btcmarkets/btcmarkets.go [hunk 5]
```diff
@@ -483,6 +488,36 @@ func (b *BTCMarkets) RemoveOrder(ctx context.Context, id string) (CancelOrderRes
 		request.Auth)
 }
 
+// Replace removes a given order
+func (b *BTCMarkets) ReplaceAnOrder(ctx context.Context, id, clientOrderID string, price, amount float64) (*OrderData, error) {
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets.go:492]: Update comment and I think this should be rn to `ReplaceOrder`

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Excellent
image
```
BTC Markets request body: {"amount":"0.001","price":"80000"}
HTTP status: 200 OK, Code: 200
BTC Markets raw response: {"orderId":"8218903930","marketId":"BTC-AUD","side":"Ask","type":"Limit","creationTime":"2022-05-08T23:28:41.131000Z","price":"80000","amount":"0.001","openAmount":"0.001","status":"Accepted","postOnly":false}
```

My main nit is why didn't anyone accept my amazing BTC offer? Please fix

**Review by gloriousCode — APPROVED** [review]
tACK!

File: exchanges/order/orders.go [hunk 6]
```diff
@@ -902,7 +902,7 @@ func StringToOrderStatus(status string) (Status, error) {
 	switch status {
 	case AnyStatus.String():
 		return AnyStatus, nil
-	case New.String(), "PLACED":
+	case New.String(), "PLACED", "ACCEPTED":
```
Comment (thrasher-) [inline exchanges/order/orders.go:905]: Please add these additions to the tests despite having test coverage happy

File: cmd/exchange_template/wrapper_file.tmpl [hunk 7]
```diff
@@ -403,7 +403,7 @@ func ({{.Variable}} *{{.CapitalName}}) SubmitOrder(ctx context.Context, s *order
 
 // ModifyOrder will allow of changing orderbook placement and limit to
 // market conversion
-func ({{.Variable}} *{{.CapitalName}}) ModifyOrder(ctx context.Context, action *order.Modify) (order.Modify, error) {
+func ({{.Variable}} *{{.CapitalName}}) ModifyOrder(ctx context.Context, action *order.Modify) (*order.Modify, error) {
```
Comment (thrasher-) [inline cmd/exchange_template/wrapper_file.tmpl:406]: return nil, common.ErrNotYetImplemented

File: go.mod [hunk 8]
```diff
@@ -24,6 +24,7 @@ require (
 	github.com/volatiletech/null v8.0.0+incompatible
 	golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4
 	golang.org/x/net v0.0.0-20220412020605-290c469a71a5
+	golang.org/x/text v0.3.7
```
Comment (thrasher-) [inline go.mod:27]: Please run go mod tidy

**Review by thrasher- — CHANGES_REQUESTED** [review]
Just three changes and this looks good to go!

image

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**